source/octf/analytics/statistics/FilesystemStatistics.cpp

```diff
@@ -8,99 +8,96 @@
-            , devId(devId)
-            , partId(partId)
-            , statsCase(sCase) {}
-
+            : Id()
+            , Size()
+            , Name("") {}
     virtual ~Key() {}
 
     Key(const Key &other)
-            : name(other.name)
-            , devId(other.devId)
-            , partId(other.partId)
-            , statsCase(other.statsCase) {}
+            : Id(other.Id)
+            , Size(other.Size)
+            , Name(other.Name) {}
 
     Key &operator=(const Key &other) {
         if (this != &other) {
-            name = other.name;
-            devId = other.devId;
-            partId = other.partId;
-            statsCase = other.statsCase;
+            Id = other.Id;
+            Size = other.Size;
+            Name = other.Name;
         }
 
         return *this;
     }
 
     bool operator==(const Key &other) const {
-        return name == other.name && devId == other.devId &&
-               partId == other.partId && statsCase == other.statsCase;
+        return Id == other.Id;
     }
 
     bool operator!=(const Key &other) const {
-        return !(*this == other);
+        return Id != other.Id;
     }
 
     bool operator<(const Key &other) const {
-        int result = name.compare(other.name);
-        if (0 == result) {
-            if (devId != other.devId) {
-                return devId < other.devId;
-            } else if (partId != other.partId) {
-                return partId < other.partId;
-            } else {
-                return statsCase < other.statsCase;
-            }
-        }
-
-        return result < 0;
+        return Id < other.Id;
     }
 
-    std::string name;
-    uint64_t devId;
-    uint64_t partId;
-    StatisticsCase statsCase;
+    uint64_t Id;
+    uint64_t Size;
+    std::string Name;
 };
 
 FilesystemStatistics::FilesystemStatistics()
-        : m_children()
-        , m_ioStats()
-        , m_devId()
-        , m_partId()
-        , m_statsCase(StatisticsCase::NAME_NOT_SET) {}
+        : m_map() {}
 
 FilesystemStatistics::~FilesystemStatistics() {}
 
 FilesystemStatistics::FilesystemStatistics(const FilesystemStatistics &other)
-        : m_children(other.m_children)
-        , m_ioStats(other.m_ioStats)
-        , m_devId(other.m_devId)
-        , m_partId(other.m_partId)
-        , m_statsCase(other.m_statsCase) {}
+        : m_map(other.m_map) {}
 
 FilesystemStatistics &FilesystemStatistics::operator=(
         const FilesystemStatistics &other) {
     if (&other != this) {
-        m_children = other.m_children;
-        m_ioStats = other.m_ioStats;
-        m_devId = other.m_devId;
-        m_partId = other.m_partId;
-        m_statsCase = other.m_statsCase;
+        m_map = other.m_map;
     }
 
     return *this;
 }
 
+void FilesystemStatistics::addDevice(
+        const proto::trace::EventDeviceDescription &devDesc) {
+    Key key(devDesc.id(), devDesc.size(), devDesc.name());
+
+    // Check maybe key already exists
+    auto iter = m_map.find(key);
```

Comment:
In what situation the id would be the same and we'd want to reinsert the device with (I'm assuming) different size and name